tensorflow/stream_executor/rocm/rocm_blas.cc

```diff
@@ -1438,8 +1442,42 @@ bool ROCMBlas::DoBlasGemm(
     float alpha, const DeviceMemory<Eigen::half> &a, int lda,
     const DeviceMemory<Eigen::half> &b, int ldb, float beta,
     DeviceMemory<Eigen::half> *c, int ldc) {
-  LOG(ERROR) << "fp16 sgemm is not implemented in this rocBLAS version";
-  return false;
+  VLOG(1) << port::Printf(
+      "doing rocBLAS SGEMM: at=%d bt=%d m=%llu n=%llu "
+      "k=%llu alpha=%f a=%p lda=%d b=%p ldb=%d beta=%f "
+      "c=%p ldc=%d",
+      static_cast<int>(transa), static_cast<int>(transb), m, n, k, alpha,
+      a.opaque(), lda, b.opaque(), ldb, beta, c->opaque(), ldc);
+  if (transa == blas::Transpose::kNoTranspose) {
+    if (lda < static_cast<int64>(m)) {
+      LOG(WARNING) << "GEMM lda was smaller than m (no transpose case); "
+                      "precondition violation";
+    }
+  } else {
+    if (lda < static_cast<int64>(k)) {
+      LOG(WARNING) << "GEMM lda (" << lda << ") was smaller than k (" << k
+                   << ") (transpose case); precondition violation";
+    }
+  }
+  if (transb == blas::Transpose::kNoTranspose) {
+    if (ldb < static_cast<int64>(k)) {
+      LOG(WARNING) << "GEMM ldb (" << ldb << ") was smaller than k (" << k
+                   << ") (no transpose case); precondition violation";
+    }
+  } else {
+    if (ldb < static_cast<int64>(n)) {
+      LOG(WARNING) << "GEMM ldb was smaller than n (transpose case); "
+                      "precondition violation";
+    }
+  }
+  const rocblas_half alpha_half = _cvtss_sh(alpha, 0);
```

Comment:
How about going the `alpha_half = Eigen::half(alpha);` routine instead? 

That would keep things consistent with the other parameters (`A`, `B`, `C`)...that way all input params need to have the same `Eigen::half` to `rocblas_half` conversion (currently just a static cast), and if there is any memory mapping we need to apply when converting `Eigen::half` to `rocblas_half`, we can apply it to all input params.

Reply:
thanks. let me give it a shot.

Reply:
`Eigen::half` has conversion operators to `unsigned short` i assume?

Reply:
Yes....the `.x` member is the unsigned short.